lib/AST/Util.cpp

```diff
@@ -75,4 +78,141 @@ void CopyProvenance(clang::Stmt *from, clang::Stmt *to, StmtToIRMap &map) {
+  }
+
+  clang::Expr *VisitFloatingLiteral(clang::FloatingLiteral *expr) {
+    LOG(INFO) << "VisitFloatingLiteral";
+    return ast.CreateFPLit(expr->getValue());
+  }
+
+  clang::Expr *VisitCastExpr(clang::CastExpr *expr) {
+    LOG(INFO) << "VisitCastExpr";
+    return ast.CreateCStyleCast(expr->getType(), Visit(expr->getSubExpr()));
+  }
+
+  clang::Expr *VisitImplicitCastExpr(clang::ImplicitCastExpr *expr) {
+    LOG(INFO) << "VisitImplicitCastExpr";
+    return Visit(expr->getSubExpr());
+  }
+
+  clang::Expr *VisitUnaryOperator(clang::UnaryOperator *expr) {
+    LOG(INFO) << "VisitUnaryOperator";
+    return ast.CreateUnaryOp(expr->getOpcode(), Visit(expr->getSubExpr()));
+  }
+
+  clang::Expr *VisitBinaryOperator(clang::BinaryOperator *expr) {
+    LOG(INFO) << "VisitBinaryOperator";
+    return ast.CreateBinaryOp(expr->getOpcode(), Visit(expr->getLHS()),
+                              Visit(expr->getRHS()));
+  }
+
+  clang::Expr *VisitConditionalOperator(clang::ConditionalOperator *expr) {
+    LOG(INFO) << "VisitConditionalOperator";
+    return ast.CreateConditional(Visit(expr->getCond()),
+                                 Visit(expr->getTrueExpr()),
+                                 Visit(expr->getFalseExpr()));
+  }
+
+  clang::Expr *VisitArraySubscriptExpr(clang::ArraySubscriptExpr *expr) {
+    LOG(INFO) << "VisitArraySubscriptExpr";
+    return ast.CreateArraySub(Visit(expr->getBase()), Visit(expr->getIdx()));
+  }
+
+  clang::Expr *VisitCallExpr(clang::CallExpr *expr) {
+    LOG(INFO) << "VisitCallExpr";
+    std::vector<clang::Expr *> args;
+    for (auto arg : expr->arguments()) {
+      args.push_back(Visit(arg));
+    }
+    return ast.CreateCall(Visit(expr->getCallee()), args);
+  }
+
+  clang::Expr *VisitMemberExpr(clang::MemberExpr *expr) {
+    LOG(INFO) << "VisitMemberExpr";
+    return clang::MemberExpr::Create(
+        ctx, expr->getBase(), expr->isArrow(), clang::SourceLocation(),
+        expr->getQualifierLoc(), clang::SourceLocation(), expr->getMemberDecl(),
+        expr->getFoundDecl(), expr->getMemberNameInfo(),
+        /*FIXME(frabert)*/ nullptr, expr->getType(), expr->getValueKind(),
+        expr->getObjectKind(), expr->isNonOdrUse());
+  }
+
+  clang::Expr *VisitDeclRefExpr(clang::DeclRefExpr *expr) {
+    LOG(INFO) << "VisitDeclRefExpr";
+    return ast.CreateDeclRef(expr->getDecl());
+  }
+
+  clang::Expr *VisitInitListExpr(clang::InitListExpr *expr) {
+    LOG(INFO) << "VisitInitListExpr";
+    std::vector<clang::Expr *> inits;
+    for (auto init : expr->inits()) {
+      inits.push_back(Visit(init));
+    }
+    return ast.CreateInitList(inits);
+  }
+
+  clang::Expr *VisitCompoundLiteralExpr(clang::CompoundLiteralExpr *expr) {
+    LOG(INFO) << "VisitCompoundLiteralExpr";
+    return ast.CreateCompoundLit(expr->getType(),
+                                 Visit(expr->getInitializer()));
+  }
+
+  clang::Expr *VisitParenExpr(clang::ParenExpr *expr) {
+    LOG(INFO) << "VisitParenExpr";
```

Comment:
Make these `LOG(INFO)` into `DLOG(INFO)`, ideally removing them eventually because I don't see why we want logging in here.